src/Post.jsx

```diff
@@ -0,0 +1,54 @@
+import React, { useState } from 'react';
+import Box from '@mui/material/Box';
+import TextField from '@mui/material/TextField';
+import Button from '@mui/material/Button';
+import SaveAltIcon from '@mui/icons-material/SaveAlt';
+import { addTrans } from './TranReducer';
+import { useDispatch, useSelector } from 'react-redux';
+
+
+export default function Post({ addSavedTransaction }) {
+  const trans = useSelector((state) => state.trans);
+  const [title, setTitle] = useState('');
+  const [date, setDate] = useState('');
+  const [category, setCategory] = useState('');
+  const [amount, setAmount] = useState('');
+  const [description, setDescription] = useState('');
+  const [nextId, setNextId] = useState(trans.length > 0 ? trans[trans.length - 1].id + 1 : 1);
+  const dispatch = useDispatch();
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    if (!title.trim() || !date.trim() || !category.trim() || !amount.trim() || !description.trim()) {
+      return;
+    }
+
+    dispatch(addTrans({ id: nextId, title, date,category,amount,description }));
+    setTitle('');
```

Comment:
you can put all these codes for clearing the form inside a separate function named **clearForm**